### ISSUE TITLE
Could not find a package configuration file provided by "cjson"

### ISSUE THREAD

**Issue by JGB-UQ** [OPM/opm-parser#651]
Hi,
I am trying to convert GRDECL file into GMSH format. I compiled the main files successfully.
Then I realized I should also install the following to be able to convert from GRDECL to GMSH:
1- OPM-Common
2- OPM-Parser
3- OPM-core
4- OPM-material

I installed cjson & boost using: 
1- sudo apt-get libjson-c-dev
2- sudo apt-get install libjson0 libjson0-dev
3- sudo apt-get install libboost-dev

I received an error message as follow :

Could not find a package configuration file provided by "cjson" with any of
  the following names:
    cjsonConfig.cmake
    cjson-config.cmake

CMake Error at CMakeLists.txt:125 (include):
  include could not find load file:

    Findopm-data

  By not providing "FindERT.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "ERT", but
  CMake did not find one.

  Could not find a package configuration file provided by "ERT" with any of
  the following names:

    ERTConfig.cmake
    ert-config.cmake

  install TARGETS given no ARCHIVE DESTINATION for static library target
  "opmjson".

CMake Error at opm/json/tests/CMakeLists.txt:1 (opm_add_test):
  Unknown CMake command "opm_add_test".

-- Configuring incomplete, errors occurred!
See also "/home/mohamed/Petrel_to_GMSH/dumux-2.8.0/opm-parser/build-cmake/CMakeFiles/CMakeOutput.log".
--- Failed to build opm-parser ---
Terminating dunecontrol due to previous errors!

Any help passing this problem or a work around so i can do the conversion in a simpler way ?

Thanks

**Comment by alfbr** [comment]
You seem to be missing one dependency, https://github.com/Ensembles/ert

**Comment by joakim-hove** [comment]
> I am trying to convert GRDECL file into GMSH format.

I am not familiar with the GMSH format - so for the end-task you are interested in I am quite ignorant, however I hope I can assist you with the build problems.

>1- OPM-Common
2- OPM-Parser
3- OPM-core
4- OPM-material

Even before that you must install ERT. After the necessary `apt`configuration (http://opm-project.org/?page_id=245) you can install ERT with
```
sudo apt-get install ert.ecl
```

or alternatively build from source: https://github.com/Ensembles/ert

>I installed cjson & boost using: 
>1- sudo apt-get libjson-c-dev
>2- sudo apt-get install libjson0 libjson0-dev

What you have installed here is *another* C library for working with json files. You should essentially **forget about** `cjson` - it will come bundled with `opm-parser` - both if you install from binary package or from source.

Looking at your error messages it can *seem* like you have not found the build configuration files from `opm-common` - have you checked out the various modules according to: http://opm-project.org/?page_id=231

**Comment by JGB-UQ** [comment]
That was really helpful. Thans a lot.
Hope you a great day.

**Comment by Mohamedss** [comment]
I followed the instructions and received the following cmake error messages:

CMake Error at /home/s4375182/Dumux/dumux-2.8.0/dune-common/cmake/modules/DuneTestMacros.cmake:286 (message):
  Please use dune_add_test instead of add_test! If you need add_test in a
  downstream project, set the variable DUNE_REENABLE_ADD_TEST to True in that
  project to suppress this error.
Call Stack (most recent call first):
  dune/alugrid/impl/test/CMakeLists.txt:44 (add_test)


CMake Error at /home/s4375182/Dumux/dumux-2.8.0/dune-common/cmake/modules/DuneTestMacros.cmake:286 (message):
  Please use dune_add_test instead of add_test! If you need add_test in a
  downstream project, set the variable DUNE_REENABLE_ADD_TEST to True in that
  project to suppress this error.
Call Stack (most recent call first):
  dune/alugrid/impl/test/CMakeLists.txt:44 (add_test)


CMake Error at /home/s4375182/Dumux/dumux-2.8.0/dune-common/cmake/modules/DuneTestMacros.cmake:286 (message):
  Please use dune_add_test instead of add_test! If you need add_test in a
  downstream project, set the variable DUNE_REENABLE_ADD_TEST to True in that
  project to suppress this error.
Call Stack (most recent call first):
  dune/alugrid/impl/test/CMakeLists.txt:44 (add_test)


CMake Error at /home/s4375182/Dumux/dumux-2.8.0/dune-common/cmake/modules/DuneTestMacros.cmake:281 (message):
  The function add_directory_test_target has been removed alongside all
  testing magic in dune-common.  Check dune_add_test for the new way!
Call Stack (most recent call first):
  dune/alugrid/impl/test/CMakeLists.txt:48 (add_directory_test_target)


-- Configuring incomplete, errors occurred!
See also "/home/s4375182/Dumux/dumux-2.8.0/dune-alugrid/build-cmake/CMakeFiles/CMakeOutput.log".
See also "/home/s4375182/Dumux/dumux-2.8.0/dune-alugrid/build-cmake/CMakeFiles/CMakeError.log".
--- Failed to build dune-alugrid ---
Terminating dunecontrol due to previous errors!

Any help ?
Thanks

**Comment by joakim-hove** [comment]
From the error Messages you seem to be using dune-control, which I am not
familiar with. I would recommend the normal cmake and make dance, that is
by far most commonly used by opm Developers.
27. des. 2015 13:38 skrev "Mohamedss" <notifications@github.com> følgende:

**Comment by atgeirr** [comment]
It looks like the errors are happening when trying to build dune-alugrid. The best source for advice on that would be to ask on the Dune mailing list (see dune-project.org).

**Comment by Mohamedss** [comment]
Thanks a lot for your reply.
Normal cmake gives me more errors, that's why I am using dune-control which works fine to some extent.

I will follow your advice and ask on the Dune mailing list hope to install it successfully as soon as possible.

Thanks again,

**Comment by JGB-UQ** [comment]
When I created two folders, one for the source and the other is the build folder I think for some reason I duplicated the files and therefore I got the following error:

Using scripts from /home/s4375182/Dumux/dumux-2.8.0/dune-common/cmake/scripts for creating doxygen stuff.
-- using /home/s4375182/Dumux/dumux-2.8.0/dune-common/doc/doxygen/Doxystyle to create doxystyle file
CMake Error at test/CMakeLists.txt:2 (add_custom_target):
  add_custom_target cannot create target "build_tests" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory
  "/home/s4375182/Dumux/dumux-2.8.0/dune-multidomaingrid".  See documentation
  for policy CMP0002 for more details.
